patches/net/minecraft/world/entity/LivingEntity.java.patch

```diff
@@ -330,9 +387,20 @@
                  this.playSound(soundtype.getFallSound(), soundtype.getVolume() * 0.5F, soundtype.getPitch() * 0.75F);
              }
          }
-@@ -1649,9 +_,9 @@
+@@ -1616,6 +_,8 @@
+         if (!(p_330394_ <= 0.0F)) {
+             int i = (int)Math.max(1.0F, p_330394_ / 4.0F);
+ 
++            net.neoforged.neoforge.common.CommonHooks.onArmorHurt(p_330843_, p_331314_, p_330394_, this);
++            p_331314_ = new EquipmentSlot[0]; //Neo: invalidates the loop.  armor damage happens in common hook
```

Comment:
Wouldn't that just cause your IDE to yell at you for unreachable code?  I don't see how that's functionally better.

Reply:
It doesn't yell at me. I guess the current code is fine. Was thinking we could just early exit 
![image](https://github.com/neoforged/NeoForge/assets/40846040/6089df17-38c8-46d8-951a-5aa5407f4153)

Reply:
Generally speaking `if (true/false) { ...` are the way that we null out code blocks we want to avoid.

The IDE will produce a warning for the code below `if (true) return;` (similar for code within an `if (false) {` block), but we don't care about warnings in the patched vanilla source, there's ~15k of them already.

Since there is no additional code below this line that needs to be run, using if (true) return;` is more expressive of the intent here (to cancel the rest of the method), so let's use that.

Reply:
changed.